pyschism/forcing/hycom/hycom2schism.py

```diff
@@ -719,112 +722,263 @@ def fetch_data(self, outdir: Union[str, os.PathLike], hgrid, vgrid, start_date,
             dst_temp['tracer_concentration'][it,:,:,0] = temp_int
 
             ds.close()
- 
-        #dst_temp.close()
-        #dst_salt.close()
-        
+
         logger.info(f'Writing *_nu.nc takes {time()-t0} seconds')
 
 class DownloadHycom:
 
-    def __init__(self, hgrid):
+    def __init__(self, hgrid=None, bbox: Bbox = None):
+        """
+        Bbox is assumed to be in lat/lon
+        """
 
-        xmin, xmax = hgrid.coords[:, 0].min(), hgrid.coords[:, 0].max()
-        ymin, ymax = hgrid.coords[:, 1].min(), hgrid.coords[:, 1].max()
-        xmin = xmin + 360. if xmin < 0 else xmin
-        xmax = xmax + 360. if xmax < 0 else xmax
-        self.bbox = Bbox.from_extents(xmin, ymin, xmax, ymax)
+        if hgrid is not None and bbox is not None:
+            raise ValueError('Arguments `hgrid` and `bbox` are mutually exclusive.')
 
-    def fetch_data(self, date, fmt='schism'):
-        '''
-        date: datetime.datetime
-        fmt: 'schism' - for Fortran code; 'hycom' - raw netCDF from HYCOM
-        '''
-
-        database=get_database(date)
-        logger.info(f'Fetching data for {date} from database {database}')
+        if hgrid is not None:
+            xy = hgrid.get_xy(crs='epsg:4326')
+            xmin, xmax = xy[:, 0].min(), xy[:, 0].max()
+            ymin, ymax = xy[:, 1].min(), xy[:, 1].max()
+            self.bbox = Bbox.from_extents(xmin, ymin, xmax, ymax)
 
+        else:
+            self.bbox = bbox
+
+    @property
+    def bbox(self):
+        return self._bbox
+
+    @bbox.setter
+    def bbox(self, bbox: Union[Bbox, None]):
+        if bbox is None:
+            bbox = Bbox.from_extents( -180., -90, 180., 90.)
+        if not isinstance(bbox, Bbox):
+            raise ValueError(f'Argument `bbox` must be of type {Bbox} or None, but got {bbox}.')
+        self._bbox = bbox
+
+    def get_dataset(
```

Comment:
Provides a "live" instance of the dataset which may be used to stream the data without incurring in IO overhead.